store/src/main/java/org/apache/rocketmq/store/ha/HAService.java

```diff
@@ -280,9 +280,14 @@ private void doWaitTransfer() {
                 if (!this.requestsRead.isEmpty()) {
                     for (CommitLog.GroupCommitRequest req : this.requestsRead) {
                         boolean transferOK = HAService.this.push2SlaveMaxOffset.get() >= req.getNextOffset();
-                        for (int i = 0; !transferOK && i < 5; i++) {
+                        long waitUntillWhen = HAService.this.defaultMessageStore.getSystemClock().now()
```

Comment:
When the sync process was notified more than 5 times, it will lead to early return before sync completely, so it seems that it should not depend on the number of notified times.